### PR TITLE
refactor(parameter-pane): implement the new parameter pane

### LINKED ISSUE #41: implement parameter pane

### PR DESCRIPTION
**Pull request by peipacut**
decided to go with a slightly different design for categories

image

[original design docs for reference](https://www.figma.com/file/nmA8maS3q92P36xTwJhO2o/physics-in-the-browser-(Copy)?type=design&node-id=501-58&mode=design&t=1hbPvzjpZghqbjJ8-0)

Currently, this PR doesn't animate hiding the parameter pane, though I don't think that should be a priority at this point.

With the code in this PR, we can introduce new parameter categories as separate function components. We can then add those components to the array in `getAllCategories` and specify whether it is an easy or expert mode option with an enum bitmask.

When reviewing, please check the way I'm selectively rendering different categories, namely the `filterOnDifficulty` and `getAllCategories` functions. I'm not sure if my method is best practice.

closes #41

### CODE REVIEW

**Comment by Lutra-Fs** [comment]
image

**Review by Lutra-Fs — CHANGES_REQUESTED** [review]
The style is broken. cc: @LorenzoValentine 
image

**Comment by LorenzoValentine** [comment]
> The style is broken. cc: @LorenzoValentine image

What do you mean by 'the style'? Do you mean that when reducing the horizontal distance of the browser, the simulation area will block the parameter pane?

File: src/components/ParametersBar.tsx [hunk 1]
```diff
@@ -1,43 +1,172 @@
...
+}) {
+  const isVisible = props.isVisible
+  const setVisible = props.setVisible
+
+  return(
+    <BackButton onClick={() => setVisible(!isVisible)}>
+      {isVisible ? (<DoubleLeftOutlined />) : (<DoubleRightOutlined />)}
+      </BackButton>
+  )
+}
```
Comment (LorenzoValentine) [inline src/components/ParametersBar.tsx:81]: Another issue is that, when you click the 'ShowHide' button, when the par bar is retracted, the position of this button does not correspond to the position of the par bar when it is displayed. Normally, they should be at the same level.
Hide:
Screenshot 2023-09-09 at 18 33 37

Display:
Screenshot 2023-09-09 at 18 33 32

**Review by LorenzoValentine — CHANGES_REQUESTED** [review]

**Comment by LorenzoValentine** [comment]
Should the font size be smaller than the normal font size?
Screenshot 2023-09-09 at 18 44 15

**Comment by peipacut** [comment]
> Should the font size be smaller than the normal font size? Screenshot 2023-09-09 at 18 44 15

image

other services like discord seem to use a similar font size for their labels and tooltips. personally, i'm okay with the size of the tooltips, making them smaller could reduce readability. thoughts? @LorenzoValentine

File: src/components/ParametersBar.tsx (hunk 1 above)
Comment (peipacut) [reply src/components/ParametersBar.tsx:82]: buttons should now be roughly the same height now

**Comment by peipacut** [comment]
new changes remove the component list idea and just conditionally renders each category. should hopefully fix any issues with `unique key prop`. also addressed the differing y positions of the expand button.

**Comment by Lutra-Fs** [comment]
@peipacut, as you are now working on the parameter pane, could you run formatting in `src/ParameterComponents` and `src\components\ParametersBar.tsx`? thanks

**Comment by LorenzoValentine** [comment]
> > Should the font size be smaller than the normal font size? Screenshot 2023-09-09 at 18 44 15
> 
> image
> 
> other services like discord seem to use a similar font size for their labels and tooltips. personally, i'm okay with the size of the tooltips, making them smaller could reduce readability. thoughts? @LorenzoValentine

But from your screenshot, the original text in Discord has a bold effect, this is different from the tooltips.

**Comment by peipacut** [comment]
latest push should resolve formatting issues.

as for the tooltips, they weren't initially bold on my end but hopefully the extra styling I added should force the tooltip text to always be unbold.

**Comment by Lutra-Fs** [comment]
> latest push should resolve formatting issues.
> 
> as for the tooltips, they weren't initially bold on my end but hopefully the extra styling I added should force the tooltip text to always be unbold.

I believe his meaning is that the original text should be different with tooltip's (in format)

**Comment by peipacut** [comment]
image

bolded the label text and left the tooltip text

**Review by Lutra-Fs — APPROVED** [review]

**Review by Lutra-Fs — APPROVED** [review]

**Review by Lutra-Fs — APPROVED** [review]

**Review by LorenzoValentine — APPROVED** [review]

**Pull request merged**